? PLX-44.patch
? src/main/org/codehaus/plexus/velocity/AbstractVelocityComponent.java
? src/main/org/codehaus/plexus/velocity/Property.java
Index: project.xml
===================================================================
RCS file: /scm/cvs/plexus/plexus-components/native/velocity/project.xml,v
retrieving revision 1.3
diff -u -r1.3 project.xml
--- project.xml	29 Oct 2003 19:01:06 -0000	1.3
+++ project.xml	20 Jan 2004 22:32:14 -0000
@@ -7,7 +7,19 @@
   <currentVersion>1.0-beta-1</currentVersion>
 
   <dependencies>
-    
+
+    <dependency>
+      <groupId>plexus</groupId>
+      <artifactId>plexus</artifactId>
+      <version>0.14-SNAPSHOT</version>
+    </dependency>
+
+    <dependency>
+      <groupId>xstream</groupId>
+      <artifactId>xstream</artifactId>
+      <version>0.3</version>
+    </dependency>
+
     <dependency>
       <groupId>commons-collections</groupId>
       <artifactId>commons-collections</artifactId>
@@ -25,7 +37,7 @@
     <dependency>
       <groupId>log4j</groupId>
       <artifactId>log4j</artifactId>
-      <version>1.2.6</version>
+      <version>1.2.8</version>
     </dependency>
 
     <dependency>
Index: src/main/org/codehaus/plexus/velocity/CommonsLoggingSystem.java
===================================================================
RCS file: /scm/cvs/plexus/plexus-components/native/velocity/src/main/org/codehaus/plexus/velocity/CommonsLoggingSystem.java,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 CommonsLoggingSystem.java
--- src/main/org/codehaus/plexus/velocity/CommonsLoggingSystem.java	6 Aug 2003 05:49:58 -0000	1.1.1.1
+++ src/main/org/codehaus/plexus/velocity/CommonsLoggingSystem.java	20 Jan 2004 22:32:14 -0000
@@ -1,8 +1,8 @@
 package org.codehaus.plexus.velocity;
 
-import java.util.Enumeration;
+//import java.util.Enumeration;
 
-import org.apache.velocity.runtime.RuntimeConstants;
+//import org.apache.velocity.runtime.RuntimeConstants;
 import org.apache.velocity.runtime.RuntimeServices;
 import org.apache.velocity.runtime.log.LogSystem;
 
Index: src/main/org/codehaus/plexus/velocity/DefaultVelocityComponent.java
===================================================================
RCS file: /scm/cvs/plexus/plexus-components/native/velocity/src/main/org/codehaus/plexus/velocity/DefaultVelocityComponent.java,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 DefaultVelocityComponent.java
--- src/main/org/codehaus/plexus/velocity/DefaultVelocityComponent.java	6 Aug 2003 05:49:58 -0000	1.1.1.1
+++ src/main/org/codehaus/plexus/velocity/DefaultVelocityComponent.java	20 Jan 2004 22:32:14 -0000
@@ -47,167 +47,79 @@
  * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  * ----------------------------------------------------------------------------
- *
  * This software consists of voluntary contributions made by many
  * individuals on behalf of the Apache Software Foundation.  For more
  * information on the Apache Software Foundation, please see
  * <http://www.apache.org/>.
- *
  * ----------------------------------------------------------------------------
  */
 
-import org.apache.avalon.framework.activity.Initializable;
-import org.apache.avalon.framework.configuration.Configurable;
-import org.apache.avalon.framework.configuration.Configuration;
-import org.apache.avalon.framework.configuration.ConfigurationException;
-import org.apache.velocity.Template;
+import org.codehaus.plexus.personality.plexus.lifecycle.phase.Initializable;
 import org.apache.velocity.app.VelocityEngine;
-import org.apache.velocity.context.Context;
-import org.apache.velocity.exception.MethodInvocationException;
-import org.apache.velocity.exception.ParseErrorException;
-import org.apache.velocity.exception.ResourceNotFoundException;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.Reader;
-import java.io.Writer;
+import java.util.List;
 
 /**
- * Description of the Class
+ * A simple velocity component implementation.
+ * 
+ * A typical configuration will look like this:
+ * <pre>
+ *      <configuration>
+ *        <properties>
+ *          <property>
+ *            <name>runtime.log.logsystem.class</name>
+ *            <value>org.codehaus.plexus.velocity.Log4JLoggingSystem</value>
+ *          </property>
+ *          <property>
+ *            <name>runtime.log.logsystem.log4j.category</name>
+ *            <value>org.codehaus.plexus.velocity.DefaultVelocityComponentTest</value>
+ *          </property>
+ *          <property>
+ *            <name>resource.loader</name>
+ *            <value>classpath</value>
+ *          </property>
+ *          <property>
+ *            <name>classpath.resource.loader.class</name>
+ *            <value>org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader</value>
+ *          </property>
+ *        </properties>
+ *      </configuration>
+ * </pre>
  */
-public class DefaultVelocityComponent
-     implements VelocityComponent, Configurable, Initializable
+public class DefaultVelocityComponent extends AbstractVelocityComponent
+     implements VelocityComponent, Initializable
 {
-    VelocityEngine engine = new VelocityEngine();
+    /** */
+    private VelocityEngine engine = new VelocityEngine();
+    
+    /** */
+    private List properties;
 
     /**
-     * Description of the Method
+     * Initializes the velocity engine.
+     * 
+     * @throws Exception Thrown if the velocity engine couldn't be initalized.
      */
-    public void configure(Configuration configuration)
-        throws ConfigurationException
+    public void initialize() 
+        throws Exception
     {
-        Configuration[] propertyElements =
-            configuration.getChild("properties").getChildren("property");
-
-        for (int i = 0; i < propertyElements.length; i++)
-        {
-            String property = propertyElements[i].getAttribute("name");
-            String value = propertyElements[i].getAttribute("value");
-            //log.debug("Property: " + property + " -> " + value);
+        for (int i = 0; i < properties.size(); i++)
+           {
+            String property = ((Property)properties.get(i)).getName();
+            String value = ((Property)properties.get(i)).getValue();
+            System.err.println("Property: " + property + " -> " + value);
             engine.setProperty(property,value);
         }
-    }
-
-    /**
-     * Description of the Method
-     */
-    public void initialize()
-        throws Exception
-    {
+        
         engine.init();
     }
 
-    // --------------------------- these methods all delegate to VelocityEngine
-
-    /**
-     * Description of the Method
-     */
-    public boolean evaluate(Context context, Writer out,
-                            String logTag, String instring)
-        throws ParseErrorException, MethodInvocationException,
-        ResourceNotFoundException, IOException
-    {
-        return engine.evaluate(context, out, logTag, instring);
-    }
-
-    /**
-     * Description of the Method
-     */
-    public boolean evaluate(Context context, Writer writer,
-                            String logTag, InputStream instream)
-        throws ParseErrorException, MethodInvocationException,
-        ResourceNotFoundException, IOException
-    {
-        return engine.evaluate(context, writer, logTag, instream);
-    }
-
-    /**
-     * Description of the Method
-     */
-    public boolean evaluate(Context context, Writer writer,
-                            String logTag, Reader reader)
-        throws ParseErrorException, MethodInvocationException,
-        ResourceNotFoundException, IOException
-    {
-        return engine.evaluate(context, writer, logTag, reader);
-    }
-
-    /**
-     * Description of the Method
-     */
-    public boolean invokeVelocimacro(String vmName, String logTag,
-                                     String params[], Context context,
-                                     Writer writer)
-        throws Exception
-    {
-        return engine.invokeVelocimacro(vmName, logTag, params,
-            context, writer);
-    }
-
-    /**
-     * Description of the Method
-     */
-    public boolean mergeTemplate(String templateName, Context context,
-                                 Writer writer)
-        throws ResourceNotFoundException, ParseErrorException,
-        MethodInvocationException, Exception
-    {
-        return engine.mergeTemplate(templateName, context, writer);
-    }
-
-    /**
-     * Description of the Method
-     */
-    public boolean mergeTemplate(String templateName, String encoding,
-                                 Context context, Writer writer)
-        throws ResourceNotFoundException, ParseErrorException,
-        MethodInvocationException, Exception
-    {
-        return engine.mergeTemplate(templateName, encoding, context, writer);
-    }
-
-    /**
-     * Gets the template attribute of the DefaultVelocityComponent object
-     */
-    public Template getTemplate(String name)
-        throws ResourceNotFoundException, ParseErrorException, Exception
-    {
-        return engine.getTemplate(name);
-    }
-
-    /**
-     * Gets the template attribute of the DefaultVelocityComponent object
-     */
-    public Template getTemplate(String name, String encoding)
-        throws ResourceNotFoundException, ParseErrorException, Exception
-    {
-        return engine.getTemplate(name, encoding);
-    }
-
     /**
-     * Description of the Method
+     * Returns the <code>VelocityEngine</code>.
+     * 
+     * @return Returns the <code>VelocityEngine</code>.
      */
-    public boolean templateExists(String templateName)
-    {
-        return engine.templateExists(templateName);
-    }
-
-    /**
-     * Sets the applicationAttribute attribute of the DefaultVelocityComponent
-     * object
-     */
-    public void setApplicationAttribute(Object key, Object value)
-    {
-        engine.setApplicationAttribute(key, value);
+    protected VelocityEngine getEngine() {
+        return engine;
     }
 }
Index: src/main/org/codehaus/plexus/velocity/Log4JLoggingSystem.java
===================================================================
RCS file: /scm/cvs/plexus/plexus-components/native/velocity/src/main/org/codehaus/plexus/velocity/Log4JLoggingSystem.java,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 Log4JLoggingSystem.java
--- src/main/org/codehaus/plexus/velocity/Log4JLoggingSystem.java	6 Aug 2003 05:49:58 -0000	1.1.1.1
+++ src/main/org/codehaus/plexus/velocity/Log4JLoggingSystem.java	20 Jan 2004 22:32:14 -0000
@@ -1,8 +1,59 @@
 package org.codehaus.plexus.velocity;
 
-import java.util.Enumeration;
+/* ----------------------------------------------------------------------------
+ * The Apache Software License, Version 1.1
+ *
+ * Copyright (c) 2000-2002 The Apache Software Foundation.  All rights
+ * reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * 3. The end-user documentation included with the redistribution, if
+ *    any, must include the following acknowlegement:
+ *       "This product includes software developed by the
+ *        Apache Software Foundation (http://www.apache.org/)."
+ *    Alternately, this acknowlegement may appear in the software itself,
+ *    if and wherever such third-party acknowlegements normally appear.
+ *
+ * 4. The names "The Jakarta Project", "Plexus", and "Apache Software
+ *    Foundation" must not be used to endorse or promote products derived
+ *    from this software without prior written permission. For written
+ *    permission, please contact apache@apache.org.
+ *
+ * 5. Products derived from this software may not be called "Apache"
+ *    nor may "Apache" appear in their names without prior written
+ *    permission of the Apache Group.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE APACHE SOFTWARE FOUNDATION OR
+ * ITS CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ * USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ * ----------------------------------------------------------------------------
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ * ----------------------------------------------------------------------------
+ */
 
-import org.apache.velocity.runtime.RuntimeConstants;
 import org.apache.velocity.runtime.RuntimeServices;
 import org.apache.velocity.runtime.log.LogSystem;
 
@@ -19,20 +70,26 @@
     /** Velocity Runtime services */
     private RuntimeServices runtimeServices = null;
 
+    /** */
     public Log4JLoggingSystem()
     {
     }
 
+    /**
+     * Initializes the logger.
+     * 
+     * @param runtimeServices The runtime services.
+     */
     public void init( RuntimeServices runtimeServices )
     {
         this.runtimeServices = runtimeServices;
     }
 
     /**
-     *  logs messages
+     * Logs messages
      *
-     *  @param level severity level
-     *  @param message complete error message
+     * @param level Severity level.
+     * @param message Complete error message.
      */
     public void logVelocityMessage(int level, String message)
     {
@@ -42,16 +99,16 @@
                 logger.warn( message );
                 break;
             case LogSystem.INFO_ID:
-                logger.info(message);
+                logger.info( message );
                 break;
             case LogSystem.DEBUG_ID:
-                logger.debug(message);
+                logger.debug( message );
                 break;
             case LogSystem.ERROR_ID:
-                logger.error(message);
+                logger.error( message );
                 break;
             default:
-                logger.debug(message);
+                logger.debug( message );
                 break;
         }
     }
Index: src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.java
===================================================================
RCS file: /scm/cvs/plexus/plexus-components/native/velocity/src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.java,v
retrieving revision 1.2
diff -u -r1.2 DefaultVelocityComponentTest.java
--- src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.java	26 Nov 2003 03:02:27 -0000	1.2
+++ src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.java	20 Jan 2004 22:32:14 -0000
@@ -16,7 +16,7 @@
 
     public void testBasic() throws Exception
     {
-        VelocityComponent velocity = (VelocityComponent) getComponent( VelocityComponent.ROLE );
+        VelocityComponent velocity = (VelocityComponent) lookup( VelocityComponent.ROLE );
 
         VelocityContext context = new VelocityContext();
 
Index: src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.xml
===================================================================
RCS file: /scm/cvs/plexus/plexus-components/native/velocity/src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.xml,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 DefaultVelocityComponentTest.xml
--- src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.xml	6 Aug 2003 05:49:58 -0000	1.1.1.1
+++ src/test/org/codehaus/plexus/velocity/DefaultVelocityComponentTest.xml	20 Jan 2004 22:32:14 -0000
@@ -1,16 +1,26 @@
 <plexus>
   <components>
     <component>
-      <logger>test</logger>
       <role>org.codehaus.plexus.velocity.VelocityComponent</role>
       <implementation>org.codehaus.plexus.velocity.DefaultVelocityComponent</implementation>
-    
       <configuration>
         <properties>
-          <property name="runtime.log.logsystem.class" value="org.codehaus.plexus.velocity.Log4JLoggingSystem"/>
-          <property name="runtime.log.logsystem.log4j.category" value="org.codehaus.plexus.velocity.DefaultVelocityComponentTest"/>
-          <property name="resource.loader" value="classpath"/>
-          <property name="classpath.resource.loader.class" value="org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader"/>
+          <property>
+          	<name>runtime.log.logsystem.class</name>
+          	<value>org.codehaus.plexus.velocity.Log4JLoggingSystem</value>
+          </property>
+          <property>
+          	<name>runtime.log.logsystem.log4j.category</name>
+          	<value>org.codehaus.plexus.velocity.DefaultVelocityComponentTest</value>
+          </property>
+          <property>
+          	<name>resource.loader</name>
+          	<value>classpath</value>
+          </property>
+          <property>
+          	<name>classpath.resource.loader.class</name>
+          	<value>org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader</value>
+          </property>
         </properties>
       </configuration>
     </component>
